src/lib/components/dropdown/Dropdown.component.js

```diff
@@ -101,86 +99,125 @@ const Caret = styled.span`
 
 const TriggerStyled = ButtonStyled.withComponent("div");
 
+const DropdownTriggerContainer = forwardRef<DropdownTriggerContainerProps, Element>(({isItem, dataIndex, open, size , variant, title, onBlur, onFocus, onClick, onMouseEnter, onMouseLeave, children, ...rest}, ref) => {
+  return isItem ? (
+    <DropdownMenuItemStyled
+      onMouseEnter={onMouseEnter}
+      onMouseLeave={onMouseLeave}
+      data-index={dataIndex}
+      onClick={onClick}
+      ref={ref}
+    >
+      {children}
+    </DropdownMenuItemStyled>
+  ) : (
+    <DropdownStyled
+      active={open}
+      variant={variant}
+      className="sc-dropdown"
+      {...rest}
+    >
+    <TriggerStyled
+      variant={variant}
+      size={size}
+      className="trigger"
+      onBlur={onBlur}
+      onFocus={onFocus}
+      onClick={onClick}
+      tabIndex="0"
+      title={title}
+      ref={ref}
+    >
+      {children}
+    </TriggerStyled>
+    </DropdownStyled>
+  )
+})
+
 function Dropdown({
+  isItem = false,
```

Comment:
I think separating exposed Dropdown component and an internal non-exposed DropdownAsAnItem component would be a better design. It would for example avoid confusion of having an `isItem` or `dataIndex`(which are implementation details) props available when trying to use the Dropdown component. I think it is very important to keep our exposed API very clear and not add props linked to the implementation in what we expose.

Reply:
done